src/common/vitals/vital-metric.js

```diff
@@ -0,0 +1,51 @@
+export class VitalMetric {
+  #values = []
+  #roundingMethod = Math.floor
+  #subscribers = new Set()
+
+  constructor (name, roundingMethod) {
+    this.name = name
+    this.attrs = {}
+    if (typeof roundingMethod === 'function') this.#roundingMethod = roundingMethod
+  }
+
+  get value () {
+    return {
+      previous: this.#values[this.#values.length - 2],
+      current: this.#values[this.#values.length - 1],
+      name: this.name,
+      attrs: this.attrs
+    }
+  }
+
+  set value (v) {
+    this.#values.push(this.#roundingMethod(v))
+    this.#subscribers.forEach(cb => {
+      try {
+        cb(this.value)
+      } catch (e) {
+        // ignore errors
+      }
+    })
+  }
+
+  get isValid () {
+    return this.value.current >= 0
```

Comment:
This should probably be `this.#values.length > 0`. Just looking to the comments down around the `cls` metric in the page view timing aggregator.